### PR TITLE
implement interactions on the to do list

### PR DESCRIPTION
**Pull request by richardoppiyo**
In this milestone:
- I implemented the changeStatus method in TODO class  to changed the status when checked
- Update item upon user keypree on index.js
- implemented clearCompleted method in the TODO class
-  worked on localStorage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,9 +27,6 @@ cleared.addEventListener('click', () => {
   newTodo.clearCompleted();
 });
 
-cleared.addEventListener('click', () => {
-  newTodo.clearCompleted();
-});
 for (let i = 0; i < statusInput.length; i += 1) {
   statusInput[i].addEventListener('change', (ev) => {
     newTodo.changeStatus(ev.target.id, ev.target.checked);
```
Comment (julie-ify) [inline src/index.js:31]: - [ ]  I can see that you have already implemented almost all the tasks for upcoming projects 🤔. Currently, the local storage does not get updated when the checkbox is changed. The required behavior is that the local storage gets updated and toggles completed from false to true and also puts a line through the completed task. Please kindly fix this by updating the local storage when there is a toggle of the checkbox and putting a line through the completed task. Please check the project requirement [here](https://www.youtube.com/watch?v=AcUd-_Yjjqg)

chrome_ULi3BxHR31

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @richardoppiyo,

Good job so far!

## Highlight!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Good code organization ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/todo.js [hunk 2]
```diff
@@ -11,8 +11,8 @@ class TODO {
 
   addTodo() {
     this.todos = this.readTodo();
-    this.description = myInput.value;
-    this.index = this.todos.length + 1;
+    this.description = descInput.value;
+    this.index = this.todos.length;
     this.todos.push({
       index: this.index,
```
Comment (brytebee) [inline src/modules/todo.js:17]: - [ ] **Optional:** The index key of your object should begin from 1 rather than 0.

**Hint** `index: this.index++`

**Review by brytebee — APPROVED** [review]
Hi @richardoppiyo 👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by richardoppiyo** [comment]
Thank you for the review. I am surely going to make the recommended changes.

**Pull request merged**